src/main/java/com/conveyal/r5/profile/PerTargetPropagater.java

```diff
@@ -0,0 +1,107 @@
+package com.conveyal.r5.profile;
+
+import com.conveyal.r5.streets.LinkedPointSet;
+import gnu.trove.map.TIntIntMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+
+/**
+ * This is similar to the Propagater class, but instead of the reducer function being called with each iteration,
+ * it is called with the travel times to a particular target for all iterations.
+ *
+ * The reducer will be called with the target index, and a boolean array of whether that target is reachable at each iteration.
+ */
+public class PerTargetPropagater {
+    private static final Logger LOG = LoggerFactory.getLogger(PerTargetPropagater.class);
+
+    /** Times at transit stops for each iteration */
+    public final int[][] travelTimesToStopsEachIteration;
+
+    /** Times at targets using the street network */
+    public final int[] nonTransferTravelTimesToTargets;
+
+    /** The maximum allowable travel time in this regional analysis, to avoid propagating things that won't be useful */
+    public final int cutoffSeconds;
+
+    /** The linked targets */
+    public final LinkedPointSet targets;
+
+    /** the profilerequest (used for walk speed etc.) */
+    public final ProfileRequest request;
+
+    public PerTargetPropagater (int[][] travelTimesToStopsEachIteration, int[] nonTransferTravelTimesToTargets, LinkedPointSet targets, ProfileRequest request, int cutoffSeconds) {
+        this.travelTimesToStopsEachIteration = travelTimesToStopsEachIteration;
+        this.nonTransferTravelTimesToTargets = nonTransferTravelTimesToTargets;
+        this.targets = targets;
+        this.request = request;
+        this.cutoffSeconds = cutoffSeconds;
+    }
+
+    public void propagate (Reducer reducer) {
+        targets.makePointToStopDistanceTablesIfNeeded();
+
+        long startTimeMillis = System.currentTimeMillis();
+        // avoid float math in loop below
+        // float math was previously observed to slow down this loop, however it's debatable whether that was due to
+        // casts, the operations themselves, or the fact that the operations were being completed with doubles rather
+        // than floats.
+        int speedMillimetersPerSecond = (int) (request.walkSpeed * 1000);
+
+        for (int targetIdx = 0; targetIdx < targets.size(); targetIdx++) {
+            boolean[] perIterationResults = new boolean[travelTimesToStopsEachIteration.length];
+            // fill with non-transit time as upper bound on travel time
```

Comment:
This is not filling it with the time, it's filling it with booleans representing whether this target cell is reachable without transit. The idea is analogous to an upper bound but I don't think we can apply the term to a boolean; it's sort of a bit pattern to be OR'ed with the transit results.